### PR TITLE
DONT MERGE, in progress: 0 2 cleanup #172

### PR DESCRIPTION
**Pull request by hendrikmuhs**
Work on #172, do not merge yet

Todo:
  - unit tests for new utils 
  - maybe: automate deprecation check on travis
  - maybe: cherry-pick deprecations to 0.1

### CODE REVIEW

File: keyvi/src/cpp/dictionary/dictionary_merger.h [hunk 1]
```diff
@@ -100,9 +100,29 @@ final {
     };
 
 public:
-    DictionaryMerger(size_t memory_limit = 1073741824, const merger_param_t& params = merger_param_t())
```
Comment (narekgharibyan) [inline keyvi/src/cpp/dictionary/dictionary_merger.h]: @hendrikmuhs 
What you think, if we change memory limit units from `byte` for `MB` ?

Comment (hendrikmuhs) [inline keyvi/src/cpp/dictionary/dictionary_merger.h]: @narekgharibyan Agreed, that makes sense.

Will not change it here, but for the new way of specifying the limit. I suggest to change the key accordingly into 'memory_limit_mb'

**Comment by hendrikmuhs** [comment]
Sorry, lot of re-formating due to re-activating cpp-lint. Most changes are format fixes.

The real changes are as follows:

a utility class t make parameter parsing a bit nicer:
https://github.com/cliqz-oss/keyvi/pull/192/files#diff-dc63a1f02927a0d37032419d691723d1

(I though back on forth: keep it simple as now or use some config parser library, .. I decided to keep it simple :-) )

using that class memory limit can be given as kb, mb. gb.

**Comment by hendrikmuhs** [comment]
I will redo from '378ecfe', not worth reviewing at the moment.

**Pull request closed without merging**